packages/astro/components/ViewTransitions.astro

```diff
@@ -97,17 +125,21 @@ const { fallback = 'animate' } = Astro.props as Props;
 		};
 
 		// Wait on links to finish, to prevent FOUC
-		const links = Array.from(doc.querySelectorAll('head link[rel=stylesheet]')).map(
-			(link) =>
-				new Promise((resolve) => {
-					const c = link.cloneNode();
+		const links: Promise<any>[] = [];
+		for(const el of doc.querySelectorAll('head link[rel=stylesheet]')) {
+			// Do not preload links that are already on the page.
+			if(!document.querySelector(`[${PERSIST_ATTR}="${el.getAttribute(PERSIST_ATTR)}"]`)) {
+				const c = document.createElement('link');
+				c.setAttribute('rel', 'preload');
+				c.setAttribute('as', 'style');
+				c.setAttribute('href', el.getAttribute('href')!);
+				links.push(new Promise<any>(resolve => {
 					['load', 'error'].forEach((evName) => c.addEventListener(evName, resolve));
```

Comment:
Out of curiosity, are these events removed when go to a new page?

Reply:
No, these are preload links and are created for new stylesheets only. I could add a `{ once: true }` but I can't think of any scenario where they'd fire again.